File: Python/Start/分割字符串的最大得分（Easy）.py
class Solution:
    def max_score(self, s: str) -> int:
        # ans = 0
        # for i in range(1, len(s)):
        #     score = 0
        #     for j in range(0, i):
        #         if s[j] == '0':
        #             score += 1
        #     for j in range(i, len(s)):
        #         if s[j] == '1':
        #             score += 1
        #     ans = max(score, ans)
        # return ans
        return max(s[:i].count('0') + s[i:].count('1') for i in range(1, len(s)))

File: Cpp/Start/分割字符串的最大得分（Easy）.cpp
#include <string>

int max(int a, int b)
{
    return a > b ? a : b;
}

class Solution 
{
public:
    int max_score(std::string s) 
    {
        int ans = 0, prev_score = 0;
        if (s[0] == '0') prev_score++;
        for (int i = 1; i < s.size(); ++i) if (s[i] == '1') prev_score++;
        ans = prev_score;
        for (int i = 1; i < s.size() - 1; ++i)
        {
            if (s[i] == '0') prev_score++;
            else prev_score--;
            ans = max(ans, prev_score);
        }
        return ans;
    }
};